emlog.c

```diff
@@ -91,6 +90,7 @@ static struct cdev *emlog_cdev = NULL;
 static struct class *emlog_class = NULL;
 static struct device *emlog_dev_reg;
 static struct emlog_info *emlog_info_list = NULL;
+static spinlock_t emlog_info_list_lock; // P714a
```

Comment:
What does P714a mean?
I suggest instead to add documentation to all global variables, which lock it is supposed to protect it.

```suggestion

static struct emlog_info *emlog_info_list = NULL;  // protected by emlog_info_list_lock
static spinlock_t emlog_info_list_lock;
```